### ISSUE TITLE
Have 'MailTrackerBlocker' display its own version number...

### ISSUE THREAD

**Issue by Mfhejjas** [apparition47/MailTrackerBlocker#73]
Please consider enhancing 'MailTrackerBlocker' to display its own version number within the drop-down dialog window when the user clicks on the ⊗ "button".  I've attached an image showing a possible way this might appear (as shown within the red ellipse)...

Regards, Mike Hejjas

[DisplayMTBversion.pdf](https://github.com/apparition47/MailTrackerBlocker/files/6121963/DisplayMTBversion.pdf)

**Comment by apparition47** [comment]
I mentioned this in #17 but I'm planning on changing the UI modal to a "popover" like the Safari one then including the version number somewhere there (maybe on the popup after pressing `ℹ️`).

![](https://user-images.githubusercontent.com/3298414/109238907-b15a5400-7817-11eb-98db-4724d2c78978.png)

**Comment by HandyMac** [comment]
As a corollary to this request: From my participation in a couple issues a few days ago, I have several "MailTrackerBlocker.pkg" installer files scattered around on my various Macs. How can I tell them apart? They show no version number in ⌘I Get Info, and they're dated only by when they were downloaded. And they don't offer to "Show Package Contents" in the contextual menu as some older .pkg files do. I opened the .pkg in Pacifist, but couldn't find a version number. 

Apparently the only way I can determine the version of a particular .pkg is to install it, then go find the installed item (as I recall – though I don't think this info is in the documentation anywhere – it's in /Library/Mail/Bundles/) and do ⌘I Get Info on it. There the version number shows; I see I have "MailTrackerBlocker.mailbundle" v.0.3.17 installed now. Kinda tedious. Suggest adding version number to the package name, e.g. "MailTrackerBlocker-0.3.21.pkg". Also in whatever is the metadata that shows in ⌘I Get Info.

**Comment by Mfhejjas** [comment]
I agree with the suggestion of incorporating the version number within the name of the .pkg file (and .dmg files as well, when used to distribute software).  Many software development organizations follow such practices already, but not all…

As for the immediate problem regarding "MailTrackerBlocker.pkg" installer file versions, one can open such a file using certain 3rd-party developer utilities, such as "Pacifist" or "Suspicious Package", then locate and open the "Info.plist" file within it, and then finally searching for the value of the "CFBundleShortVersionString" key within that.  Definitive, but not ideal, particularly for any non-technical users…

Regards,

Mike Hejjas

**Comment by apparition47** [comment]
0.4.1 now displays its version number from the new popover.

As for including some versioning in the pkg itself--that's TODO

**Comment by Mfhejjas** [comment]
An email message can contain more than one kind of tracker, can it not?  And yet, for every email I've received so far that 'MailTrackerBlocker' reported containing a named tracker, apparently each of those emails only contained a single one.  Does MTB only report the first tracker detected, or does it report them all?  If MTB doesn't report all, please consider enhancing it to do so...

Regards,

Mike Hejjas

**Comment by apparition47** [comment]
I also thought email trackers would be like website trackers--as in, there would be multiple trackers from different vendors for each page/email. But since the past year, I've come across **only one** email that had more than one tracker and from separate vendors (CNN Newsletters #93). This seems to be an edge case. For this specific case, MailTrackerBlocker reports only the first but blocks all.

**Comment by m-schmitt** [comment]
> As a corollary to this request: From my participation in a couple issues a few days ago, I have several "MailTrackerBlocker.pkg" installer files scattered around on my various Macs. How can I tell them apart? They show no version number in ⌘I Get Info, and they're dated only by when they were downloaded. And they don't offer to "Show Package Contents" in the contextual menu as some older .pkg files do. I opened the .pkg in Pacifist, but couldn't find a version number.

This is why I always rename downloaded installers with the version number.

Since you have Pacifist, you can find the version number in Info.plist > CFBundlePackageType. In the latest package this is in MailTrackerBlocker > Core Contents > MailTrackerBlocker.mailbundle> Contents > Info.plst.
